### ISSUE TITLE
[BUG] avoid-redundant-async doesn't handle nullable return values

### ISSUE THREAD

**Issue by maxlapides** [dart-code-checker/dart-code-metrics#1006]
* **Dart code metrics version: 4.19.0**
* **Dart sdk version: 2.18.1**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
dart_code_metrics:
  rules:
    - avoid-redundant-async
```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart
Future<void> foo() async {
  return maybeBar?.foo();
}
```

**What did you expect to happen?**
This should be allowed by the lint rule because the `async` keyword is not redundant in this case. This function may return either `null` or a `Future` (assuming `maybeBar.foo()` returns a `Future`).

**What actually happened?**
Lint warning for `avoid-redundant-async`

**Are you willing to submit a pull request to fix this bug?**
Maybe

**Comment by incendial** [comment]
Available in 4.19.1 🚀

**Comment by maxlapides** [comment]
Thank you @incendial!!!

**Comment by incendial** [comment]
> Thank you @incendial!!!

Thanks for reporting the problem!
